### PR TITLE
LRQA-67888 - Create test stubs for Workflow Metrics - Performance By Step

### PR DESCRIPTION
**Pull request by evanilson-santana**

### CODE REVIEW

**Comment by anabuchmann** [comment]
ci:test:sf

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+	@description = "LPS-100674 - Verify that the card will update the step name if it is changed in Process Builder"
+	@ignore = "Test Stub"
+	@priority = "4"
+	test CardUpdatesStepNameWhenStepNameIsChangedInProcessBuilder {
+
+		// TODO LRQA-67676 CardUpdatesStepNameWhenStepNameIsChangedInProcessBuilder pending implementation
+
+	}
+
+	@description = "LPS-100674 - Verify that the card (Performance by Steps) works as designed when the portal is set to a non-en-US locale"
```
Comment (anabuchmann) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase:76]: Please, remove the parenthesis from (Performance by Step).
When, in the test map, you have double quotation marks in the test case description, you can only remove them from the stub description.

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+	@description = "LPS-100674 - Verify that when the user selects Last 7 Days time period, the result and filter options displayed in accordance with reference"
+	@ignore = "Test Stub"
+	@priority = "4"
+	test FilterItemsFromLast7Days {
+
+		// TODO LRQA-67687 FilterItemsFromLast7Days pending implementation
+
+	}
+
+	@description = "LPS-100674 - Verify that when the user selects (Last 90 Days) time period the result and filter options displayed in accordance with reference"
```
Comment (anabuchmann) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase:112]: Please, remove the parenthesis from (Last 90 Days).
When, in the test map, you have double quotation marks in the test case description, you can only remove them from the stub description.

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+	@ignore = "Test Stub"
+	@priority = "4"
+	test FilterItemsFromLast90Days {
+
+		// TODO LRQA-67685 FilterItemsFromLast90Days pending implementation
+
+	}
+
+	@description = "LPS-100674 - Verify that when the user selects (Last 180 Days) time period the result and filter options are displayed in accordance with reference"
```
Comment (anabuchmann) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase:121]: Please, remove the parenthesis from (Last 180 Days).
When, in the test map, you have double quotation marks in the test case description, you can only remove them from the stub description.

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+	@ignore = "Test Stub"
+	@priority = "4"
+	test FilterItemsFromLast180Days {
+
+		// TODO LRQA-67684 FilterItemsFromLast180Days pending implementation
+
+	}
+
+	@description = "LPS-100674 - Verify that when the user selects (Last Year) time period the result and filter options displayed in accordance with reference"
```
Comment (anabuchmann) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase:130]: Please, remove the parenthesis from (Last Year).
When, in the test map, you have double quotation marks in the test case description, you can only remove them from the stub description.

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+	test FilterPerformanceByStepPagePerTimeRange {
+
+		// TODO LRQA-67667 FilterPerformanceByStepPagePerTimeRange pending implementation
+
+	}
+
+	@description = "LPS-98233 - Verify that the fields From and To are required"
+	@ignore = "Test Stub"
+	@priority = "3"
+	test FromAndToAreRequiredFields {
```
Comment (anabuchmann) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase:160]: Please, in description, change the LPS number for 100674.

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+
+		// TODO LRQA-67686 FromAndToAreRequiredFields pending implementation
+
+	}
+
+	@description = "LPS-100674 - Verify that hovering the info icon next to (Performance by Step) card will display a tooltip"
```
Comment (anabuchmann) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase:166]: Please, remove the parenthesis from (Perfomance by Step).
When, in the test map, you have double quotation marks in the test case description, you can only remove them from the stub description.

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+	@description = "LPS-100677 - Verify the page is providing loading feedback when the table is being updated according to the search queries"
+	@ignore = "Test Stub"
+	@priority = "2"
+	test PageProvidesLoadingFeedbackOnTableUpdate {
+
+		// TODO LRQA-67670 PageProvidesLoadingFeedbackOnTableUpdate pending implementation
+
+	}
+
+	@description = "LPS-100677 - Verify that the search is an approximate match (aka (LIKE) search type or (character by character comparison))"
```
Comment (anabuchmann) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase:202]: Please, remove the parenthesis from (LIKE) and (character by character comparison).
When, in the test map, you have double quotation marks in the test case description, you can only remove them from the stub description.

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+	@description = "LPS-97667 - Verify that when there are tasks with average time smaller than 1 minute the average completion time is displayed"
+	@ignore = "Test Stub"
+	@priority = "4"
+	test TasksWithAverageTimeSmallerThan1MinAreDisplayed {
+
+		// TODO LRQA-67668 TasksWithAverageTimeSmallerThan1MinAreDisplayed pending implementation
+
+	}
+
+	@description = "LPS-97667 - Verify that, when displaying time, the time units are shortened by using the biggest measurement possible, and use up to 2 units per time, with (minute) as smallest value and (days) as biggest unit"
```
Comment (anabuchmann) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase:220]: Please, remove the parenthesis from (minute) and (days).
When, in the test map, you have double quotation marks in the test case description, you can only remove them from the stub description.

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase [hunk 9]
```diff
@@ -0,0 +1,274 @@
...
+	@description = "LPS-100677 - Verify that searching steps containing non-ascii chars is possible"
+	@ignore = "Test Stub"
+	@priority = "3"
+	test UserCanSearchNonAsciiChars {
+
+		// TODO LRQA-67675 UserCanSearchNonAsciiChars pending implementation
+
+	}
+
+	@description = "LPS-100674 - Verify that the card (Performance by Step), when it has 6 or more entries, the table should provide a vertical scroll"
```
Comment (anabuchmann) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase:247]: Please, remove the parenthesis from (Performance by Step)
When, in the test map, you have double quotation marks in the test case description, you can only remove them from the stub description.

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase [hunk 10]
```diff
@@ -0,0 +1,274 @@
...
+	@ignore = "Test Stub"
+	@priority = "2"
+	test WhenCard6OrMoreEntriesTableProvideVerticalScroll {
+
+		// TODO LRQA-67679 WhenCard6OrMoreEntriesTableProvideVerticalScroll pending implementation
+
+	}
+
+	@description = "LPS-100674 - Verify that the card (Performance by Step), when it has 5 or fewer entries, the table should not scroll"
```
Comment (anabuchmann) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase:256]: Please, remove the parenthesis from (Performance by Step)
When, in the test map, you have double quotation marks in the test case description, you can only remove them from the stub description.

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase [hunk 11]
```diff
@@ -0,0 +1,274 @@
...
+	@ignore = "Test Stub"
+	@priority = "2"
+	test WhenCardHasUpTo5EntriesTableDoNotScroll {
+
+		// TODO LRQA-67677 WhenCardHasUpTo5EntriesTableDoNotScroll pending implementation
+
+	}
+
+	@description = "LPS-100674 - Verify that when an SLA is deleted, the data in (Performance by Step) is still part of the average calculation"
```
Comment (anabuchmann) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testFunctional/tests/WorkflowMetricsPerformanceByStep.testcase:265]: Please, remove the parenthesis from (Performance by Step)
When, in the test map, you have double quotation marks in the test case description, you can only remove them from the stub description.

**Review by anabuchmann — CHANGES_REQUESTED** [review]
Hi @evanilson-santana !
Could you, please, check my comments and make the changes that I requested?
Thank you!

**Comment by rodrigocunhaa** [comment]
Just started reviewing :)

**Comment by rodrigocunhaa** [comment]
Hi @evanilson-santana ,

Thank you for your commit.

There was only one LPS number to fix, but I did it here: https://github.com/rodrigocunhaa/liferay-portal/pull/441

**Pull request closed without merging**